src/media/js/controller/weather-controller.js

```diff
@@ -0,0 +1,51 @@
+(function(){
+  "use strict";
+
+  angular
+    .module('WeatherApp')
+    .controller('WeatherController', WeatherController);
+
+  WeatherController.$inject = ['$http'];
+
+  function WeatherController($http) {
+    var vm = this;
+    vm.forecast = null;
+    vm.loadingMessage = "Retrieving weather details";
+    vm.getTemperature = getTemperature;
+
+    function getTemperature() {
```

Comment:
For this task - I think it was ok just to do everything in the controller. 

EBI pattern (clean architecture) would have been preferred for larger projects.